src/js/basic/solverBasic.ts

```diff
@@ -0,0 +1,109 @@
+//import { getById, showSolution, visible } from '../utilities';
+
+export function basic_solve(): void {
+	const sol: HTMLDivElement = <HTMLDivElement>document.querySelector("#BasicWires div.solution"); // Could be a utility that takes the ID.
+
+	const colour = {
+		white: "white",
+		yellow: "yellow",
+		blue: "blue",
+		red: "red",
+		black: "black"
+	};
+
+	const cut = {
+		last: "last",
+		first: "first",
+		second: "2nd",
+		fourth: "4th",
+		third: "3rd"
+	};
+
+	const getcount = (c: string): number => document.querySelectorAll(`.basicwires_${c}:checked`).length;
+
+	let count = {
+		red: getcount(colour.red),
+		blue: getcount(colour.blue),
+		yellow: getcount(colour.yellow),
+		white: getcount(colour.white),
+		black: getcount(colour.black),
+		total: 0
+	};
+
+	const lastwire = () => {
+		let w: NodeListOf<HTMLInputElement> = document.querySelectorAll('.basicwires:checked');
+		return w[w.length - 1].value;
+	}
+
+	// Todo: https://dirask.com/posts/TypeScript-sum-subtract-numbers-in-array-with-reduce-D9W7Zj
+	count.total = count.red + count.blue + count.yellow + count.black + count.white;
+
+	if (count.total < 3) {
+		sol.innerHTML = "";
+		return;
+	}
+
+	let result = [];
+	switch (count.total) {
+		case 3: {
+			if (count.red == 0) {
+				result.push(cut.second);
+			} else if (lastwire() == colour.white) {
+				result.push(cut.last);
+			} else if (count.blue > 1) {
+				result.push(`${cut.last} ${colour.blue}`);
+			} else {
+				result.push(cut.last);
+			}
+			break;
+		}
+		case 4: {
+			if (count.red > 1) {
+				result.push(`Serial odd: ${cut.last} ${colour.red}. Else: `);
```

Comment:
I think maybe I was wrong to say it should work like this. It should probably have buttons for the serial after all.
Half for consistency and half for if there's more than one basic wires module.

Reply:
We originally talked about having global buttons up the top for battery count etc, but then figured it wasn't worth it because some questions might not need to be asked. But maybe we should after all. 
E.g. a specific battery count rather than only the questions the modules ask.